capstone-rs/src/instruction.rs

```diff
@@ -385,6 +388,44 @@ impl<'a> Display for Instructions<'a> {
         Ok(())
     }
 }
+/// Registers read and/or written to by a given instruction
+#[derive(Debug)]
+pub struct RegAccess {
+    /// `Vec` containing the `RegId` of the registers read from
+    read: Vec<RegId>,
```

Comment:
we can avoid the 2 heap allocations by storing the `cs_regs` in this struct along with their lengths:

```rust
pub struct RegAccess {
    regs_read: cs_regs,
    regs_read_count: u8,
    regs_write: cs_regs,
    regs_write_count: u8,
}
```

You can add methods to access the data safely as a slice reference.

See `InsnDetail::regs_read()` in `instruction.rs` for an example.